test.html

```diff
@@ -0,0 +1,63 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Red Squares Grid</title>
+    <link rel="stylesheet" href="test.css">
+    <link rel="preconnect" href="https://fonts.googleapis.com">
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
+    <link href="https://fonts.googleapis.com/css2?family=Fira+Mono:wght@400;500;700&display=swap" rel="stylesheet">
+</head>
+<body>
+    <div class="background-layer"></div>
+    <div class="grid-container">
+        <!-- 24 squares (8x3) -->
+        <div class="grid-cross-item"></div>
+        <div></div>
+        <div class="grid-fill-item"></div>
+        <div></div>
+        <div></div>
+        <div></div>
```

Comment:
This is unavoidable I guess, since you want the grid layout... 
Not sure if adding a dummy class called like 'bg-transparent' or something to each blank div adds readability or adds dead code (probably the latter)